text/0012-corner-smoothing.md

```diff
@@ -0,0 +1,266 @@
+
+At a high level, we're taking the points where the edges connect to the corner circle and stretching them back into the edge. Thus, instead of constructing a round corner with a quarter circle, we use two cubic Bézier curves to construct the corner.
+
+![A chart showing two constructions of a rounded corner. The first construction shows a quarter circle connected to two straight edges, with the points of connection highlighted. Arrows by those points indicate a stretching motion twoard the straight edge. The second construction shows the same corner with the connection points stretched backward, demonstrating the Bézier construction of a rounded corner.](../images/0012/Construction.svg)
+
+The two Bézier curves' control points are carefully chosen to retain the most important qualities of a rounded corner, including the percieved radius. The original article provides a full intuition for how the curve's control points are derived.
+
+This method provides some creative control in choosing how far along the edge the corner's curve can extend into.
+
+### Interactions & Corner Cases
+
+<!-- Haha, "corner" cases. -->
+
+#### Simple Rounding vs Smooth Rounding
+
+Applying smoothing to an existing round corner should retain as much of the original shape's important points and properties as possible. The smoothing procedure chosen satisfies a few important mathematical constraints:
+
+* At the points where the box's edge meets the curve:
+  * The curvature is zero, creating a seamless transition from edge to corner.
+* At the point where the two curves meet:
+  * The curvature is equal, continuing the "roundness" of the corner seamlessly.
+  * The position is the same as the "middle" point in simple rounding.
+
+#### Accessory Shapes (Borders, Outlines, Shadows)
+
+More than just the corners of the element itself, `border-radius` also applies to other shapes, including borders, outlines, and shadows. These other shapes are derived from the element's shape. `-electron-corner-smoothing` also applies to these same shapes to maintain consistency.
+
+<!--
+This is the technical portion of the RFC. Explain the design in sufficient detail that:
+
+- Its interaction with other features is clear.
+- It is reasonably clear how the feature would be implemented.
+- Corner cases are dissected by example.
+- Any new dependencies on Chromium code are outlined.
+
+The section should return to the examples given in the previous section, and explain more fully how
+the detailed proposal makes those examples work.
+-->
+
+## Drawbacks
+
+Electron rarely makes changes to Blink. The few exceptions include exposing the `<webview>` element, exposing the `-webkit-app-region` CSS rule, and increasing the threshold for Web Storage APIs. With this feature, we would maintain a medium-sized patch for a feature *addition*.
```

Comment:
The `webviewTag` webpreferences toggle determines whether the webview element is available in pages. I think we should consider how we could make this CSS property be similarly opt-in.

If the opt-in mechanic were per-document, we might also be able to expand it for other properties like `-webkit-app-region`. As an aside, I've been thinking about whether we could choose to disable that in arbitrary webpages where they maybe shouldn't allow dragging the window.

Another reason to make this opt-in would be to avoid introducing a vector for browser fingerprinting.